### ISSUE TITLE
auto-brb has many problems

### ISSUE THREAD

**Issue by slingamn** [oragono/oragono#518]
I'm just brain-dumping here, don't mind me.

Auto-brb has problems:

1. The trigger (seeing an error from `socket.Read()` on the main client goroutine) is also activated by destroying a session on ping timeout: therefore, ping timeout races against auto-brb
1. Reattach after brb (auto or otherwise) does not replay history or announce to clients, leading to a situation where the client may lose history (and be aware of it, because the client sees its own disconnection and reconnection) while friends remain unaware.

**Comment by slingamn** [comment]
The quit message after failure to return from BRB should probably add a "3m30s ago" label for the benefit of friends.

**Comment by slingamn** [comment]
Getting some doubled snomasks:

````
@time=2019-05-26T03:33:42.209Z :irc.darwin.network NOTICE operator :-QUIT- edmobile exited the network
@time=2019-05-26T03:33:42.225Z :irc.darwin.network NOTICE operator :-QUIT- edmobile exited the network
````

**Comment by slingamn** [comment]
`KILL` doesn't work in production right now under some circumstances (resume? auth? brb?); unclear whether this will be resolved after merging #525 completely

**Comment by slingamn** [comment]
KILL (and other operations that do a `client.destroy(nil)`) are not effective on BRB'ed clients, this needs to be fixed.
